### ISSUE TITLE
Voice Chat

### ISSUE THREAD

**Issue by 0xfaulty** [ddnet/ddnet#6547]
**I would like to discuss the possibility of adding this feature.**

Yes, I know there's discord, but I don't always want to add everyone just to have ability to respond by voice or play several game rounds with voice (with random players party).
No, I'm not suggesting to do our own voice chat implementation entirely, we could use some free to use library and open source codec.

**Reason for adding:**
I was recently discussing with an acquaintance why there are no modifications in the game that would allow the implementation of games like "mafia" or "among us" and similar and came to the conclusion that there are two main problems for this
1. all clients have the ability to turn off textures
2. the game has no verbal communication, which is necessary for the modifications described above

If the first problem can be (partially?) solved by sending the player's position only where other can see it, the second is more global...

**Terms and conditions:**
- It should be optional feature for server
- It should be disabled by default to not confuse new players
- It could be disabled personally like mute players client feature
- It should be accepted that it is impossible to fully moderate this type of communication, so we should not save audio streams
- Server should be able to control who hears who for each player, to provide game moments
- Sound should not be common to the whole server, but within a given radius and teams separated, to avoid situations where 64 players (or more) are talking at the same time

I repeat that I am only considering this at the moment as part of the solution to the problems with adding the modifications described above. Will this be needed on ddrace servers? Not sure.

**Why I don't see the point of making a separate client with voice chat:**
It's quite hard to support own client, so if there is no way to add this to ddnet, there is generally no point in even trying, time has shown that branching clients only live long when they have cheats in them..

I would be happy to hear your views on this, and perhaps also your suggestions on how to solve the above problems.

**Comment by Kaffeine** [comment]
For a different mod we also need a voice chat with support for

> Server should be able to control who hears who for each player, to provide game moments

IIUC it is not possible for e.g. Discord to move users between voice chat channels, I also can't find that feature in Jitsi or Mumble or other voice chat system, thought I'm sure there is a software capable doing that.

It can be possible to simulate that using positional audio, e.g. teeworlds server can tell the voice chat server that players A, B, C are at the same position here, and players D, E, F are in a different and very far position there, so those two groups (teams) can't hear each other, despite of being in the same voice chat.

**Comment by heinrich5991** [comment]
I don't really like the idea of having a separate voice chat because of the moderation issues. We currently fail to moderate the text chat in a sane way; until this is fixed, I don't think voice chat would be a good idea.

**Comment by Kaffeine** [comment]
Do we *really* need to moderate team chats?
Players join a team and lock it. Probably they're OK to speak with each-other and do not need extra moderation.

**Comment by Learath2** [comment]
I'm not even sure if it's viable to ever enable this on official servers. We host them on fairly weak machines that and at about 50kbps per person it might be a little too much for the mediocre uplinks most of our hosters have.

**Comment by Robyt3** [comment]
I don't think this is a good idea. We don't have the resources to moderate voice chat, yet would have to obligation to do so for stores like Google Play. At most I might support using some API to forward the positional information to an external audio tool.

**Comment by Robyt3** [comment]
Not planned due the impossibility of moderation and the additional network load.

**Comment by ChillerDragon** [comment]
Could also ship android client without voice chat.
